File: services/consultationservice/internal/model/comment.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Comment struct {
	ID              primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	PostID          string             `bson:"post_id" json:"post_id"`
	UserID          string             `bson:"user_id" json:"user_id"`
	Content         string             `bson:"content" json:"content" validate:"required,max=500"`
	ParentCommentID string             `bson:"parent_comment_id,omitempty" json:"parent_comment_id,omitempty"`

	
	Depth           int                `bson:"depth" json:"depth"`                     
	Path            string             `bson:"path,omitempty" json:"path,omitempty"`   
	ReplyCount      int                `bson:"reply_count" json:"reply_count"`         

	IsDeleted       bool               `bson:"is_deleted" json:"is_deleted"`
	CreatedAt       time.Time          `bson:"created_at" json:"created_at"`
	UpdatedAt       time.Time          `bson:"updated_at" json:"updated_at"`

	
	Replies         []Comment          `bson:"-" json:"replies,omitempty"`             
	AuthorID        string             `bson:"-" json:"author_id,omitempty"`           
	Author          *AuthorInfo        `bson:"-" json:"author,omitempty"`              
}

type AuthorInfo struct {
	ID        string `json:"id"`
	FirstName string `json:"first_name"`
	LastName  string `json:"last_name"`
	Avatar    string `json:"avatar,omitempty"`
}

File: services/mobileflutter/mobileapp/lib/models/comment.dart
class Comment {
  final String id;
  final String postId;
  final String authorId;
  final String? authorName;
  final String? authorAvatar;
  final String content;
  final String? parentCommentId;
  final String createdAt;
  final List<Comment> replies;

  Comment({
    required this.id,
    required this.postId,
    required this.authorId,
    this.authorName,
    this.authorAvatar,
    required this.content,
    this.parentCommentId,
    required this.createdAt,
    required this.replies,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    var rawReplies = json['replies'] as List<dynamic>? ?? [];
    List<Comment> parsedReplies = rawReplies.map((r) => Comment.fromJson(r)).toList();

    return Comment(
      id: json['id'] ?? '',
      postId: json['post_id'] ?? '',
      authorId: json['author_id'] ?? '',
      authorName: json['author_name'],
      authorAvatar: json['author_avatar'],
      content: json['content'] ?? '',
      parentCommentId: json['parent_comment_id'],
      createdAt: json['created_at'] ?? '',
      replies: parsedReplies,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'post_id': postId,
      'author_id': authorId,
      'author_name': authorName,
      'author_avatar': authorAvatar,
      'content': content,
      'parent_comment_id': parentCommentId,
      'created_at': createdAt,
      'replies': replies.map((r) => r.toJson()).toList(),
    };
  }
}
